src/analysis/bounds.rs

```diff
@@ -153,14 +160,8 @@ impl Bounds {
                         });
                     }
                 }
-                if (!need_is_into_check || !*par.nullable)
-                    && par.c_type != "GDestroyNotify"
-                    && !self.add_parameter(&par.name, &type_string, bound_type, r#async)
-                {
-                    panic!(
-                        "Too many type constraints for {}",
-                        func.c_identifier.as_ref().unwrap()
-                    )
+                if (!need_is_into_check || !*par.nullable) && par.c_type != "GDestroyNotify" {
+                    self.add_parameter(&par.name, &type_string, bound_type, r#async)
```

Comment:
Good change, it took me some time to find this call when skimming through the code. Now it's more clear :)

Reply:
Yup, this just panics now when running out of type parameters to use :tada: